pulsar-broker/pom.xml

```diff
@@ -308,14 +308,14 @@
       <groupId>${project.groupId}</groupId>
       <artifactId>pulsar-functions-api-examples</artifactId>
       <version>${project.version}</version>
-      <scope>test</scope>
+      <scope>provided</scope>
```

Comment:
the reason for this is to get it off the classpath. There shouldn't be a need to have it on the classpath. 
the reason to have the `provided` scope is to add a project dependency between this module and pulsar-functions-api-examples. `maven-dependency-plugin` is used to copy the `pulsar-functions-api-examples` jar file to the build directory.

Reply:
actually when thinking more about it, changing from `test` to `provided` won't take it off the classpath as I have intended. I guess I should also change the type to `pom` so that there's a dependency to the project (so that `-am` works), but nothing gets added to the classpath. I'll revisit this.

Reply:
I now pushed a fix which reverts the change from test to provided, but instead changes the type to `pom` so that the classes aren't added to the classpath at all. 

If the classes are on the classpath, it wouldn't be possible to test loading the functions from a file or URI since the classes would always get loaded from the parent classloader which would contain the classes. That is one of the rationals to make the changes in this PR.